tileserver/__init__.py

```diff
@@ -264,68 +268,77 @@ def handle_request(self, request):
         coord = request_data.coord
         format = request_data.format
 
-        tile_data = self.reformat_from_stored_json(request_data, layer_data)
-        if tile_data is not None:
-            return self.create_response(
-                request, 200, tile_data, format.mimetype)
-
-        tile_size = request_data.tile_size
-        meta_coord, offset = self.coord_split(coord, tile_size)
-
-        if self.using_metatiles():
-            # make all formats when making metatiles
-            wanted_formats = self.formats
-
-        else:
-            wanted_formats = [json_format]
-            # add the request format, so that it gets created by the tile
-            # render process and will be saved along with the JSON format.
-            if format != json_format:
-                wanted_formats.append(format)
-
-        nominal_zoom = coord.zoom
-        cut_coords = []
-        if self.using_metatiles():
-            nominal_zoom = meta_coord.zoom + self.metatile_zoom
-            if self.metatile_zoom > 0:
-                cut_coords.extend(
-                    coord_children_range(meta_coord, nominal_zoom))
-
-        # fetch data for all layers, even if the request was for a partial set.
-        # this ensures that we can always store the result, allowing for reuse,
-        # but also that any post-processing functions which might have
-        # dependencies on multiple layers will still work properly (e.g:
-        # buildings or roads layer being cut against landuse).
-        unpadded_bounds = coord_to_mercator_bounds(meta_coord)
-        feature_data_all = self.data_fetcher(
-            nominal_zoom, unpadded_bounds, self.layer_config.all_layers)
-
-        formatted_tiles_all, extra_data = process_coord(
-            meta_coord,
-            nominal_zoom,
-            feature_data_all['feature_layers'],
-            self.post_process_data,
-            wanted_formats,
-            feature_data_all['unpadded_bounds'],
-            cut_coords, self.buffer_cfg)
-
-        expected_tile_count = len(wanted_formats) * (1 + len(cut_coords))
-        assert len(formatted_tiles_all) == expected_tile_count, \
-            'unexpected number of tiles: %d, wanted %d' \
-            % (len(formatted_tiles_all), expected_tile_count)
-
-        if layer_spec == 'all':
-            tile_data = self.extract_tile_data(
-                coord, format, formatted_tiles_all)
-
-        else:
-            # select the data that the user actually asked for from the
-            # JSON/all tile that we just created.
-            json_data_all = self.extract_tile_data(
-                coord, json_format, formatted_tiles_all)
-
-            tile_data = reformat_selected_layers(
-                json_data_all, layer_data, coord, format, self.buffer_cfg)
+        with self.cache.lock(coord):
```

Comment:
Should this include the `format` argument as well? (i.e: `self.cache.lock(coord, format)`). Same for `self.cache.get` and `set` below?